cosmwasm/contracts/crosschain-registry/Cargo.toml

```diff
@@ -0,0 +1,67 @@
+# This feature has been stabilized in 1.64. If using an earlier version (like in
+# rust-optimizer-arm:0.12.8) you should uncomment this like to be able to
+# compile the contract
+#
+#cargo-features = ["workspace-inheritance"]
+
+[package]
+name = "crosschain-registry"
+version = "0.1.0"
+authors = ["Adam Tucker <adam@osmosis.team>"]
+edition = "2021"
+
+exclude = [
+  # Those files are rust-optimizer artifacts. You might want to commit them for convenience but they should not be part of the source code publication.
+  "contract.wasm",
+  "hash.txt",
+]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[lib]
+crate-type = ["cdylib", "rlib"]
+
+[features]
+# for more explicit tests, cargo test --features=backtraces
+backtraces = ["cosmwasm-std/backtraces"]
+# use imported feature to disable all instantiate/execute/query exports
+imported = []
+
+[package.metadata.scripts]
+optimize = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer:0.12.11
+"""
+optimize-m1 = """docker run --rm -v "$(pwd)":/code \
+  --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+  --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+  cosmwasm/rust-optimizer-arm64:0.12.11
+"""
+
+[[bin]]
+name = "build-schema"
+path = "build-schema/schema.rs"
+
+[dependencies]
+cosmwasm-schema =  { workspace = true }
+cosmwasm-std =  { workspace = true }
+cosmwasm-storage = { workspace = true }
+cw-storage-plus = { workspace = true }
+cw2 = { workspace = true }
+schemars = { workspace = true }
+osmosis-std = { workspace = true }
+serde = { workspace = true }
+thiserror = { workspace = true }
+serde-json-wasm = { workspace = true }
+serde-cw-value = { workspace = true }
```

Comment:
do we need all of these dependencies? Can we remove the ones we don't need?

Reply:
Removed the ones I think were not needed https://github.com/osmosis-labs/osmosis/pull/4320/commits/ff227cffe240a9c7bd52660439a162e203ca2ea2

Is there a better way to do this other than just removing, saving, and seeing if there are any errors? 😅

Reply:
There are external tools for this. You could use [cargo udeps](https://github.com/est31/cargo-udeps) or, my new favourite because of the name, [cargo machete](https://blog.benj.me/2022/04/27/cargo-machete/)